database/dbHelpers.js

```diff
@@ -39,12 +39,79 @@ const insertFollow = (follow, done) => {
   client.create(record, (err, resp) => {
     if(err) { console.log(err); }
     else {
-      console.log('record inserted successfully, record:', resp);
+      console.log('record inserted successfully, record: ', resp);
       done();
+      trans.end();      
+    }
+  });
+}
+
+const getFeed = (user, done, trans) => {
+  let { user_id } = user;
+  client.search({
+    index: 'follows',
+    q: 'follower_id:' + user_id
+  }, (err, resp) => {
+    if(err) { console.log(err); }
+    else {
+      sortFollows(resp.hits.hits, done, trans);
     }
   })
 }
 
+const sortFollows = (follows, done, trans) => {
+  follows.sort((a, b) => {
+    let aEngagement = parseFloat(a._source.engagement_rating);
+    let bEngagement = parseFloat(b._source.engagement_rating);
+
+    if(aEngagement - bEngagement > 0) {
+      return -1;
+    } else if (aEngagement - bEngagement < 0) {
+      return 1;
+    }
+  });
+  getFeedAuthors(follows, done, trans);
+};
+
+const getFeedAuthors = (follows, done, trans) => {
+  let top5FollowRelationships = follows.slice(0, 5);
+  let usersToBeInFeed = top5FollowRelationships.map((follow) => follow._source.followed_id);
+  getTweetsForFeed(usersToBeInFeed, done, trans);
+};
+
+const getTweetsForFeed = (users, done, trans) => {
+  let userIds = users.map(user=> user).join(' OR ');
+  client.search({
+    index: 'slimtweets',
+    type: 'tweet',
+    q: 'user_id: ' + userIds,
+    size: 10000
+  }, (err, resp) => {
+    if(err) { console.log(err); }
+    else {
+      finalizeFeed(resp.hits.hits, done, trans)
+      
+    }
+  })
+};
+
+const finalizeFeed = (tweets, done, trans) => {
+  tweets.sort((a, b) => {
+    let aTime = new Date(a._source.created_at);
+    let bTime = new Date(b._source.created_at);
+
+    if(aTime < bTime) {
+      return -1;
+    } else if(aTime > bTime) {
+      return 1;
+    }
+  });
+
+  let result = tweets.slice(0, 200);
```

Comment:
If you are able to sort in the db query, then you might be able to cut it off at 200 then as well.

Reply:
Additionally, one way I've trimmed down my system is to pull 100 tweets instead of 200 for now. We should be consistent--I don't mind changing, but 100 seemed reasonable.